### PR TITLE
repository layer. 

### PR DESCRIPTION
**Pull request by 7adidaz**

### CODE REVIEW

File: src/api/reactions/reactionRepository.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const reactionRepository = {
+	addReaction: async (reaction: CreateReactionDto): Promise<Reaction> => {
+		const ids = await db('reactions').insert(reaction);
+		const newReaction = await db('reactions').where('id', ids[0]).first();
+		return newReaction;
+	},
+	getReactionById: async (id: number): Promise<Reaction> => {
+		return await db('reactions').where('id', id).first();
+	},
```
Comment (7adidaz) [inline src/api/reactions/reactionRepository.ts:13]: do we want this?

**Review by 7adidaz — COMMENTED** [review]
please look at the comments

File: src/api/messages/messageModel.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// content string
+//   sender_id uuid fk
+//   channel_id uuid fk
+//   parent_message_id uuid fk
+//   workspace_id uuid fk
+export const BaseMessageSchema = z.object({
+	content: z.string(),
+	senderId: z.number(),
+	channelId: z.number(),
+	workspaceId: z.number(),
```
[the remark this answers is not in the document]
Comment (AbdelrahmanEldabsha) [reply src/api/messages/messageModel.ts:12]: i found it in the schema so i left it

File: src/api/workspace/__tests__/02_workspaceRepository.tes.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			name: 'workspace5',
+			description: 'description',
+			avatarUrl: 'url',
+		});
+		expect(workspace.id).not.toBeNull();
+		const workspaces = await trx.select('*').from('workspaces');
+		expect(workspaces).toHaveLength(5);
+		// await EntityFactory.deleteWorkspaces([workspace.id]);
+	});
+
```
[the remark this answers is not in the document]
Comment (7adidaz) [reply src/api/workspace/__tests__/02_workspaceRepository.tes.ts:35]: ok, fix it, please.

**Pull request merged**